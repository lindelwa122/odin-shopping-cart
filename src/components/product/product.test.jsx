import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import Product from './product';

describe('Product Component', () => {
  const product = (() => {
    const getId = () => 'shop-1';
    const getName = () => 'Scarf';
    const getPrice = () => 78;
    const getDescr = () => 'A beautiful red scarf.';
    const getImg = () => 'https://fake-api.img.com/shop-1';

    return { getId, getName, getPrice, getDescr, getImg };
  })();

  it('renders correctly', () => {
    const { container } = render(<Product product={product}  />);
    expect(container).toMatchSnapshot();
  });
});