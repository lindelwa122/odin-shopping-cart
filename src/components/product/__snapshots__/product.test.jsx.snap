// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Product Component > renders correctly 1`] = `
<div>
  <div>
    <div>
      <img
        alt="Scarf"
        src="https://fake-api.img.com/shop-1"
      />
    </div>
    <div>
      <p
        aria-label="product name"
      >
        Scarf
      </p>
      <p
        aria-label="product price"
      >
        R
        78
      </p>
    </div>
  </div>
</div>
`;
